kolibri/core/assets/src/state/actions.js

```diff
@@ -223,6 +223,36 @@ function getCurrentSession(store) {
   .catch(error => { handleApiError(store, error); });
 }
 
+function getFacilityConfig(store) {
+  const coreApp = require('kolibri');
+  const FacilityCollection = coreApp.resources.FacilityResource
+    .getCollection()
+    .fetch();
+
+  return FacilityCollection.then(facilities => {
+    const currentFacilityId = facilities[0].id;
+    const facilityConfigCollection = coreApp.resources.FacilityDatasetResource
+      .getCollection({ facility_id: currentFacilityId })
+      .fetch();
+    return facilityConfigCollection
+      .then(facilityConfig => {
+        const config = {
+          id: facilityConfig[0].id,
+          description: facilityConfig[0].description,
+          location: facilityConfig[0].location,
+          learnerCanSignUp: facilityConfig[0].learner_can_sign_up,
+          learnerCanEditUsername: facilityConfig[0].learner_can_edit_username,
+          learnerCanEditName: facilityConfig[0].learner_can_edit_name,
+          learnerCanEditPassword: facilityConfig[0].learner_can_edit_password,
+          learnerCanDeleteAccount: facilityConfig[0].learner_can_delete_account,
+        };
+        store.dispatch('CORE_SET_FACILITY_CONFIG', config);
+      })
+      .catch(error => handleApiError(store, error));
+  }).catch(error => handleApiError(store, error));
```

Comment:
Since the error handler for both the inner and outer promise look the same, you could remove the `catch` one line 251.